frontend/viewer/src/lib/components/reorderer/reorderer-trigger.svelte

```diff
@@ -0,0 +1,38 @@
+<script lang="ts" module>
+  import type {Snippet} from 'svelte';
+  import type {HTMLAttributes} from 'svelte/elements';
+
+  type TriggerHTMLAttributes = HTMLAttributes<HTMLElement> & {
+    id?: string;
+  };
+
+  export type ReordererTriggerProps = TriggerHTMLAttributes & {
+    child?: Snippet<[{ arrowIcon: IconClass, props: TriggerHTMLAttributes }]>;
+  };
+</script>
+
+<script lang="ts">
+  import {Icon} from '$lib/components/ui/icon';
+  import * as DropdownMenu from '$lib/components/ui/dropdown-menu';
+  import {pickIcon} from './icon-util';
+  import {buttonVariants} from '../ui/button';
+  import type {IconClass} from '$lib/icon-class';
+  import {useReordererTrigger} from './reorderer.svelte';
+
+  let {
+    child,
+    ...props
+  } : ReordererTriggerProps = $props();
+
+  const itemListState = useReordererTrigger();
+  const {first, last, direction} = $derived(itemListState);
+  const arrowIcon = $derived(pickIcon(direction, first, last));
```

Comment:
similar issues here, changing the direction after the component is made does nothing